drivers/Makefile.dep

```diff
@@ -323,13 +323,18 @@ ifneq (,$(filter lc709203f,$(USEMODULE)))
   FEATURES_REQUIRED += periph_gpio_irq
 endif
 
+ifneq (,$(filter lis2dh12_spi,$(USEMODULE)))
+  FEATURES_REQUIRED += periph_gpio
+  FEATURES_REQUIRED += periph_spi
+else
+  FEATURES_REQUIRED += periph_i2c
```

Comment:
With this `periph_i2c` will now *always* be used.

Reply:
Ups :dark_sunglasses: yes indeed. Good catch. Thanks.

Reply:
Then we really need a nested dependency if we want an interface selected by default when only the `lis2dh12` module is enabled:
```
ifneq (,$(filter lis2dh12_int,$(USEMODULE)))
  FEATURES_REQUIRED += periph_gpio_irq
endif

ifneq (,$(filter lis2dh12%,$(USEMODULE)))
  USEMODULE += lis2dh12
  ifneq (,$(filter lis2dh12_spi,$(USEMODULE)))
    FEATURES_REQUIRED += periph_gpio
    FEATURES_REQUIRED += periph_spi
  else
    FEATURES_REQUIRED += periph_i2c
  endif
endif
```

Reply:
I changed this back to nested.